src/test/java/dev/cdevents/CDEventsTest.java

```diff
@@ -215,6 +336,33 @@ void createTaskRunFinishedEventAsCloudEvent() {
         assertThat(ceDataJson).isEqualTo(cdEventJson);
     }
 
+    @Test
+    void createTaskRunFinishedEventWithOptionalFieldsUnset() throws IOException {
+        InputStream inputStream = getClass().getResourceAsStream("/taskrun_finished_optional.json");
+        TaskRunFinishedCDEvent expectedCDEvent = objectMapper.readValue(inputStream, TaskRunFinishedCDEvent.class);
```

Comment:
I think this can be ok, it guarantees consistent behaviour, however, I'm not 100% it guarantees that optional fields are not included in the rendered JSON.

At this level, I know that:
- if I create an object without optional fields, those optional fields are not set in the object.
- if I render that object without optional fields and parse it back,  those optional fields are not set in the object.
- If a load an object from JSON that does not specify optional fields, those optional fields are not set in the object

I'm still not 100% sure that the rendered JSON will not contain the optional fields set as well, but perhaps that's covered already by other tests - wdyt? 

I see two possible ways we could test that further - possibly not for every event, just once might be enough.
- either read bytes from `inputStream` and compare it with `cdEventJson` converted to bytes. This can be tricky since you may need to compact the json first (to avoid issues with spaces, indentations etc) and also ordering in dicts may be an issue unless the json rendered implementation always renders in the same order
- or add one resource file with the optional fields set to empty value. Load it and check that the optional fields are in the object, with empty or null value.

Reply:
Yes it compares the Subject's content object as whole but not field to filed comparison.
Or I can compare from both the objects that all optional fields are set to null and they are equal, something like;
```
assertThat(cdEvent.getSubject().getContent().getTaskName()).isEqualTo(null);
assertThat(expectedCDEvent.getSubject().getContent().getTaskName()).isEqualTo(null);
```

Reply:
Now updated the test validation by comparing the JsonNodes created from `inputStream `and `cdEventJson `
This way I can compare actual and expected optional and mandatory fields from the test data.